src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -1057,7 +1133,12 @@ private void Walk(string context, Action walk)
         /// </summary>
         private bool IsReference(IOpenApiReferenceable referenceable)
         {
-            return referenceable.Reference != null && !_inComponents;
+            var isReference = referenceable.Reference != null && !_inComponents;
+            if (isReference)
+            {
+                Walk(referenceable);
+            }
```

Comment:
This Walk should be done by each of the callers. It's a bit weird to have IsReference function do something rather than just returning the boolean.

Reply:
I agree, however lifting this to the calling function would cause all the code to be lifted.  I renamed it to ProcessAsReference instead.  If the element is processed as a Reference then it will return true and no further processing of the element will happen.  By renaming the function we avoid the nastiness of having a isXXX method have a sideeffect.

Reply:
I see. This new naming sounds good to me.

---
In reply to: [259667050](https://github.com/Microsoft/OpenAPI.NET/pull/381#discussion_r259667050) [](ancestors = 259667050)